crates/fuzzing/src/oracles.rs

```diff
@@ -1055,3 +1055,73 @@ fn set_fuel<T>(store: &mut Store<T>, fuel: u64) {
     // double-check that the store has the expected amount of fuel remaining
     assert_eq!(store.consume_fuel(0).unwrap(), fuel);
 }
+
+/// Generate and execute a `crate::generators::component_types::TestCase` using the specified random bytes to drive
+/// creation of arbitrary types and values.
+#[cfg(feature = "component-model")]
+pub fn dynamic_component_api_case(bytes: &[u8]) -> arbitrary::Result<()> {
```

Comment:
Currently the general style of this crate is that the `oracles` here don't take `Unstructured` but rather take a list of inputs. I think that would be  possible to do here? This would take a `TestCase` as input as well as a `Vec<(...)>` for inputs/outputs of each run.

Reply:
I'm not sure how to do that, given that we can't create an instance of `Val` without first getting the `Type`s from a `Func` from an `Instance`, and the fuzzer has no way to do that.

Reply:
I guess I do know a way -- I used to have a `Value` enum in `generator::component_types` which could be constructed independently and then later converted to a `Val` when the `Type` was available.  I had removed that because it was redundant in the current approach, but I could restore it.

Reply:
@alexcrichton How strongly do you feel about this?  I think it would require adding a substantial amount of code.

Reply:
Oh right sorry yes I forgot that we need the component on-hand to generate the values. Yeah in that case the refactor isn't worth it and the current structure is good 👍